public/src/Helpers.ts

```diff
@@ -231,6 +231,7 @@ export const setStatusAll = (submission: Submission, status: Submission_Status):
 
 /** getCourseID returns the course ID determined by the current route */
 export const getCourseID = (): bigint => {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
Instead of ignoring this particular one I think we should convert `getcourseID` to a proper hook.

We could move it to a separate file under `/public/hooks`:


```ts
import { useParams } from "react-router"


export const useCourseID = (): bigint => {
    const route = useParams<{ id?: string }>()
    return route.id ? BigInt(route.id) : BigInt(0)
}
```